nexus/social_media/api.py

```diff
@@ -0,0 +1,82 @@
+from django.utils import timezone
+
+# Third Party Stuff
+from rest_framework import viewsets
+from rest_framework import mixins
+from rest_framework.decorators import action, parser_classes
+from rest_framework.parsers import FormParser, MultiPartParser
+from rest_framework.permissions import IsAdminUser, IsAuthenticated
+
+from nexus.base import response
+from . import models, serializers, permissions
+
+
+class PostViewSet(mixins.ListModelMixin, mixins.CreateModelMixin,
+                  mixins.UpdateModelMixin, mixins.RetrieveModelMixin,
+                  mixins.DestroyModelMixin, viewsets.GenericViewSet):
+    queryset = models.Post.objects.all()
+    permission_classes = (IsAuthenticated, permissions.IsAdminOrAuthorOfPost)
+
+    def get_serializer_class(self):
+        if self.action in ('approve', 'posted'):
+            return serializers.AdminPostSerializer
+        return serializers.PostSerializer
+
+    @action(methods=['PATCH'], detail=True, permission_classes=[IsAdminUser])
+    def approve(self, request, pk=None):
+        instance = self.get_object()
+        if not instance.is_approved:
+            data = {'is_approved': True, 'approval_time': timezone.now()}
+            return self.update_post_status(instance, data)
+        else:
+            return self.update_post_status(instance)
+
+    @action(methods=['PATCH'], detail=True, permission_classes=[IsAdminUser])
+    def posted(self, request, pk=None):
+        instance = self.get_object()
+        if not instance.is_approved:
+            return response.BadRequest({'error_message': 'Post has not been approved yet'})
+        if not instance.is_posted:
+            data = {'is_posted': True, 'posted_time': timezone.now()}
+            return self.update_post_status(instance, data)
+        else:
+            return self.update_post_status(instance)
+
+    @action(methods=['POST'], detail=True)
+    @parser_classes((FormParser, MultiPartParser))
+    def upload_image(self, request, pk=None):
+        instance = self.get_object()
+        if request.FILES:
+            if not instance.image == '':
+                return response.BadRequest({'error_message': 'Image already present'})
+            return self.update_post_status(instance, data=request.data)
+        return response.BadRequest({'error_message': 'Image file missing from the request'})
+
+    @action(methods=['POST'], detail=True)
+    @parser_classes((FormParser, MultiPartParser))
+    def update_image(self, request, pk=None):
+        instance = self.get_object()
+        if request.FILES:
+            if instance.image == '':
```

Comment:
Not correct way to check this.

Reply:
I've refactored it with a better statement.